### ISSUE TITLE
[Bug]: No longer works with default push settings

### ISSUE THREAD

**Issue by icco** [AndreasAugustin/actions-template-sync#573]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Starting with 2.3, the following config fails:

```
name: template sync
on:
  # run every day at midnight
  schedule:
    - cron: '0 0 * * *'
  # manually trigger the workflow
  workflow_dispatch:
jobs:
  repo-sync:
    runs-on: ubuntu-latest
    permissions:
      contents: write
      pull-requests: write
    steps:
      - name: Checkout
        uses: actions/checkout@v4
        with:
          token: ${{ secrets.INFRA_SYNC_TOKEN }}
      - name: actions-template-sync
        uses: AndreasAugustin/actions-template-sync@v2
        with:
          github_token: ${{ secrets.INFRA_SYNC_TOKEN }}
          source_repo_path: "pinginc/template-infra"
          source_repo_ssh_private_key: ${{ secrets.INFRA_TEMPLATE_SYNC_TOKEN }}
          is_pr_cleanup: true
```

### Expected Behavior

I expected this to work as it did in 2.2.3

### Current Behavior

The error that occurs is:

```
  ::info::push changes
  error: src refspec origin does not match any
  error: failed to push some refs to ' --set-upstream'
```

### Steps To Reproduce

Any configuration that does not end up with additional flags for `git push`

### Possible Solution

Rewrite the function  at https://github.com/AndreasAugustin/actions-template-sync/blob/7cd796fa1c887cb432d6a1d37e78172320131953/src/sync_template.sh#L290-L309

### Template sync version Version

2.4.0

**Comment by icco** [comment]
Created https://github.com/AndreasAugustin/actions-template-sync/pull/574 to fix

**Comment by AndreasAugustin** [comment]
Hi @icco and thanks a lot for the report and the PR!
Will look ASAP into it

**Comment by hdesai-dave** [comment]
we are also blocked by this

**Comment by AndreasAugustin** [comment]
> we are also blocked by this

@hdesai-dave 
:information_source: you can always temp mitigate the issue with pining the issue to the last working version. E.g.

```yaml
      - name: actions-template-sync
        uses: AndreasAugustin/actions-template-sync@v2.2.3
        with:
```

**Comment by AndreasAugustin** [comment]
Like @icco already mentioned within the description this change is responsible: 
https://github.com/AndreasAugustin/actions-template-sync/commit/01f7623dbfc8f73ccaf1635d64cb629a699acc9a#diff-2c34df082012b54923698ceb27ff084102dc0cbe7f71ce907b4a8ba41657c6edL289

**Comment by AndreasAugustin** [comment]
further PR with a test improvement #577

**Comment by AndreasAugustin** [comment]
@icco thanks a lot for your PR. The issue should be solved with new version `v2.4.1`
